diff --git a/zlibstubs.c b/zlibstubs.c
index a627df9..d962e57 100644
--- a/zlibstubs.c
+++ b/zlibstubs.c
@@ -168,7 +168,7 @@ value camlzip_inflateEnd(value vzs)
 
 value camlzip_update_crc32(value crc, value buf, value pos, value len)
 {
-  return caml_copy_int32(crc32((uint32) Int32_val(crc), 
+  return caml_copy_int32(crc32((uint32_t) Int32_val(crc), 
                           &Byte_u(buf, Long_val(pos)),
                           Long_val(len)));
 }
